### ISSUE TITLE
Qualify SemIf/MLX as a decision-native sidecar for Bithkuil research workflows

### ISSUE THREAD

**Issue by crates** [entif-ai/bithkuil#65]
## Summary

Qualify a local, decision-native scoring sidecar for bounded Bithkuil research decisions on the M3 Ultra, using SemIf's MLX path as the first reference implementation.

This is an engineering accelerator for candidate triage and curriculum control. It is not a truth oracle, a source of gold labels, or a student initializer.

SemIf is a useful first candidate because it exposes typed option probabilities directly from model logits, supports state-prefix reuse and parallel shared-state decisions, and has a native MLX backend for macOS arm64.

## Target uses

Use decision-native scoring only where the task is already a bounded choice over explicit alternatives.

Good first candidates include:

- #53 candidate relation-family classification;
- #53 candidate mapping disposition proposals such as accepted/provisional/ambiguous/rejected/deferred;
- #16 choice among already-admissible curriculum stages;
- review-priority scoring for unresolved mappings;
- binary or small-enum checks used to route records toward deterministic validation or human review.

Do not force open-ended semantic discovery into a fixed-choice interface merely to use this tool.

## Authority boundary

The sidecar MAY:

- rank or score explicit candidate options;
- select among options already permitted by deterministic policy;
- flag low-margin or high-entropy decisions for review;
- reduce generative overhead for repetitive classification.

The sidecar MUST NOT:

- emit authoritative gold labels;
- alter oracle semantics;
- certify source truth;
- access SEALED evaluation;
- initialize student embeddings or transfer hidden states into the student;
- silently promote its probability into calibrated truth confidence;
- bypass #7/#8/#9 semantic validation or #35 source/rights policy.

Every accepted mapping or curriculum action still passes the existing deterministic/formal boundary.

## SemIf reference path

Initial qualification should pin:

- exact SemIf commit/revision;
- exact upstream model revision;
- tokenizer/runtime versions;
- prompt/state rendering hash;
- MLX version;
- decision mode: fresh, serial-prefix, or shared-state;
- optional 8-bit/4-bit MLX quantization as separate measured profiles;
- allocator cache limit.

SemIf's own documentation warns that MLX's inactive allocation cache can retain substantial unified memory. When the sidecar shares the M3 Ultra with Bithkuil training, cap and record the allocator cache rather than letting a helper process starve the student.

## Benchmark against the current generative path

Use a frozen owned fixture derived from real Bithkuil mapping/curriculum decisions.

Compare decision-native scoring with the current generative local-pedagogue path on:

- decision accuracy/agreement against deterministic or human-audited labels where such labels exist;
- abstention/review routing;
- wall time;
- decisions/sec;
- generated output tokens;
- peak unified memory;
- prompt/prefix reuse benefit;
- choice changes under quantization;
- calibration metrics where enough labeled examples exist.

Do not treat agreement with the generative teacher as ground truth.

## Shared-state opportunities

Exploit SemIf's strongest systems shape when many bounded questions share one semantic state.

Examples:

```text
one candidate lexical item
  + many typed relation questions

one curriculum-state snapshot
  + many admissibility/priority criteria

one semantic graph neighborhood
  + several candidate edge classifications
```

This should reduce repeated context processing if the local workload actually has that shape.

## Optional comparators

Nimble, Decider, or another Jev-style implementation may be added later if a compatible Apple-Silicon path and licensing posture justify the extra work.

They are not prerequisites for the SemIf qualification.

## Experimental-contamination rule

If a decision-native sidecar is used only for pre-SYS corpus/graph construction, record its provenance in the resulting artifacts.

If it participates in adaptive curriculum during SYS-01, #22 must freeze its exact identity, privileges, fallback behavior, and call budget before efficacy results are visible.

Changing the helper midtrial creates a new plan/deviation epoch.

## Acceptance criteria

- [ ] SemIf or an equivalent decision-native engine runs locally through MLX on the M3 Ultra.
- [ ] A frozen Bithkuil-owned decision fixture exists for evaluation.
- [ ] At least one #53 or #16 bounded-decision path can consume typed scores without granting truth authority.
- [ ] Exact model/runtime/prompt identities are recorded.
- [ ] Fresh versus prefix-reuse/shared-state performance is measured where applicable.
- [ ] 4-bit/8-bit profiles, if used, are evaluated separately from source precision.
- [ ] Peak unified memory and allocator-cache policy are recorded.
- [ ] Low-margin/unsupported choices can route to review rather than forced acceptance.
- [ ] No sidecar output is used as a student embedding or gold label.
- [ ] #35 records license/source posture for SemIf code and upstream model assets before committed use.
- [ ] Any SYS-01 use is frozen through #22 before execution.

## Dependencies

#35 source/licensing policy and the Phase 0 local-machine profile from #52. Full #52 pilot completion is not required.

## Consumers

#16 developmental teacher and #53 lexical/graph bootstrap.

Parent roadmap: #5.

## Workability

`EXECUTABLE_LEAF`: bounded local systems experiment with one clear authority boundary and measurable performance/quality outputs.
